tensorflow/python/keras/mixed_precision/experimental/autocast_variable_test.py

```diff
@@ -157,25 +163,25 @@ def test_method_delegations(self, distribute):
     # Test AutoCastVariable correctly delegates Variable methods to the
     # underlying variable.
     with get_distribute_scope(distribute):
-      evaluate = self.evaluate
       for read_dtype in (dtypes.float32, dtypes.float16):
+        evaluate = partial(self.check_and_evaluate, dtype=read_dtype)
         x = get_var(7., dtypes.float32)
         x = autocast_variable.create_autocast_variable(x)
         with ops.get_default_graph()._enable_auto_casting_variables(
             read_dtype):
-          evaluate(x.initializer)
-          self.assertEqual(evaluate(x.value()), 7)
-          self.assertEqual(evaluate(x.read_value()), 7)
+          self.evaluate(x.initializer)
+          self.assertEqual(self.evaluate(x.value()), 7)
+          self.assertEqual(self.evaluate(x.read_value()), 7)
           self.assertTrue(x.trainable)
           self.assertEqual(x.synchronization, x._variable.synchronization)
           self.assertEqual(x.aggregation, x._variable.aggregation)
-          self.assertEqual(evaluate(x.initialized_value()), 7)
+          self.assertEqual(self.evaluate(x.initialized_value()), 7)
           if not context.executing_eagerly():
             if not distribute:
               # These functions are not supported for DistributedVariables
               x.load(9)
               self.assertEqual(x.eval(), 9)
-            self.assertEqual(evaluate(x.initial_value), 7)
+            self.assertEqual(self.evaluate(x.initial_value), 7)
             self.assertEqual(x.op, x._variable.op)
             self.assertEqual(x.graph, x._variable.graph)
           if not distribute:
```

Comment:
You should also test that you can assign the return value of `x.assign(8)`. And that if you pass `read_value=False`, the return value is an Operation or None instead of a variable.

I would move the `assign`, `assign_add`, and `assign_sub` calls to a new test method as well.

Reply:
:+1: I added the tests in c87a0e20d31cccd70a0f1c9687221b45066786fa